### PR TITLE
Add support for different tag operations

### LINKED ISSUE #12: Add support for different tag operations
Add support for three tag operations on edit:

1- replace: Remove all existing tags from the entity and just add the newly sent ones
2- append: Add the newly sent tags to the entity but do not touch the existing tags
3- remove: Just remove the sent tags from the entity but leave the rest

### PR DESCRIPTION
**Pull request by ercanozkaya**
Closes #12

### CODE REVIEW

**Comment by ercanozkaya** [comment]
@johanjanssens Can you review the code please?

@rmdstudio See description at #12 
Can you test the functionality a bit please?
docman/code/administrator/components/com_docman/views/documents/tmpl/modal_batch.html.php:25 has a hidden input.
- If the value is set to append (as it is now) it will just add new tags and keep the old ones.
- If you change the value to remove and send a tag name it will remove the tag from the entity but keep the rest.
- And when you go to single document edit form it should replace tags with the current ones sent in the form.

File: controller/behavior/taggable.php [hunk 1]
```diff
@@ -49,41 +49,67 @@ protected function _setTags(KControllerContextInterface $context)
...
                     } else {
                         $status = null;
                     }
 
-                    //Delete tags
-                    if(count($tags))
-                    {
-                        $tags->delete();
-                        $tags->reset();
+                    $tags      = $entity->getTags();
```
Comment (johanjanssens) [inline controller/behavior/taggable.php]: @ercanozkaya Could we clean this up a bit more, think you can have 3 seperate methods, 

_removeTags
_replaceTags
_appendTags

Each method would receive the entity collection and handle it. That makes much more clear what the code does?

**Comment by ercanozkaya** [comment]
@johanjanssens Can you review again please?

File: controller/behavior/taggable.php [hunk 2]
```diff
@@ -43,85 +46,112 @@ protected function _setTags(KControllerContextInterface $context)
...
                     $operation = $entity->tags_operation;
 
-                    if (!in_array($entity->tags_operation, array('append', 'replace', 'remove'))) {
-                        $operation = 'replace';
+                    if ($operation === 'remove') {
+                        $this->_removeTags($entity);
+                    } else if ($operation === 'append') {
+                        $this->_appendTags($entity);
                     }
+                    else $this->_replaceTags($entity);
```
Comment (johanjanssens) [inline controller/behavior/taggable.php]: @ercanozkaya Some code styling here.

File: controller/behavior/taggable.php [hunk 3]
```diff
@@ -43,85 +46,112 @@ protected function _setTags(KControllerContextInterface $context)
...
+
+    /**
+     * Removes the tags sent in the request from the entity
+     * @param KModelEntityInterface $entity
+     */
+    protected function _removeTags(KModelEntityInterface $entity)
+    {
+        $tags = $entity->getTags();
+
+        foreach ($tags as $tag) {
```
Comment (johanjanssens) [inline controller/behavior/taggable.php]: @ercanozkaya Some code styling here, new line after ) for opening {

**Comment by johanjanssens** [comment]
@ercanozkaya Added feedback, only some code styling issues left.

**Comment by ercanozkaya** [comment]
@johanjanssens Fixed the styling issues. Merging to master and Kodekit.

**Pull request merged**